### PR TITLE
Lift -triton-raise-block-pointer pass from intel-xpu-backend-for-triton

### PR DESCRIPTION
**Pull request by rolfmorel**
Code was in turn taken from triton-shared (though does not use the tts dialect).

### CODE REVIEW

File: third_party/cpu/include/TritonRaiseBlockPointer/Passes.td [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef TRITON_RAISE_BLOCK_POINTER_PASSES
+#define TRITON_RAISE_BLOCK_POINTER_PASSES
+
+include "mlir/Pass/PassBase.td"
+
+def TritonRaiseBlockPointer
+    : Pass<"triton-raise-block-pointer", "mlir::ModuleOp"> {
```
Comment (adam-smnk) [inline third_party/cpu/include/TritonRaiseBlockPointer/Passes.td:15]: nit: `triton-cpu-`

**Comment by adam-smnk** [comment]
Atm it doesn't really compose with triton-cpu rewrites in the way I'd expect.
When input kernel has hand written block ptrs, `triton-cpu-convert-memory-ops` manages to translate `tt.load` into 2D `vector.transfer_read`. After raising from scalar loads, the loads get split into unrolled vector instructions so the source memref is lost at this point.

**Comment by adam-smnk** [comment]
Actually `canonicalize` undoes the raising back into `tt.splat` and `tt.addptr` so imo the raising is pretty poor.

**Comment by adam-smnk** [comment]
Ohhh now I see looking closer at IR. Created `tt.make_tensor_ptr` is unused. No wonder it gets removed.

**Comment by adam-smnk** [comment]
> Ohhh now I see looking closer at IR. Created `tt.make_tensor_ptr` is unused. No wonder it gets removed.

After an offline discussion, we found the issue. Raising fails to kick in for a kernel with fully `constexpr` arguments. It does seem to work on unmodified `03-gemm` example.

File: third_party/cpu/include/TritonRaiseBlockPointer/Passes.td (hunk 1 above)
Comment (rolfmorel) [reply third_party/cpu/include/TritonRaiseBlockPointer/Passes.td]: Done.

**Comment by rolfmorel** [comment]
Rebased, added `-cpu` to pass name (and added MLIR arg names to kernel args in `03-matrix-multiplication-cpu.py`) and made the raising parameterized.

The raising indeed seems to work on unmodified `03-gemm` but appears a bit fragile.

File: python/tutorials/03-matrix-multiplication-cpu.py [hunk 2]
```diff
@@ -328,7 +335,7 @@ def matmul(a: torch.Tensor, b: torch.Tensor, c: torch.Tensor):
 
 torch.manual_seed(0)
 
-triton.runtime.driver.set_active_to_cpu(experimental=True)
+triton.runtime.driver.set_active_to_cpu(experimental=False)
```
[the remark this answers is not in the document]
Comment (rolfmorel) [reply python/tutorials/03-matrix-multiplication-cpu.py:338]: With `experimental=True` the V2 backend is used for the correctness check. We modify the V1 pipeline. Note that the correctness check only runs once, for one backend/pipeline. 

My mistake for not highlighting the change - I forgot. Nevertheless, I think we should change this (a separate PR would be more appropriate, but I don't think we need to spend time on all that).

**Comment by rolfmorel** [comment]
Indeed, no changes to the raising logic. Just moving some code over and slotting in some utility functions.

File: python/tutorials/03-matrix-multiplication-cpu.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (adam-smnk) [reply python/tutorials/03-matrix-multiplication-cpu.py:338]: Ohhh that totally makes sense and explains why it's been so slow recently.
Let's absolutely leave it changed to V1.

**Review by adam-smnk — APPROVED** [review]

File: python/tutorials/03-matrix-multiplication-cpu.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (adam-smnk) [reply python/tutorials/03-matrix-multiplication-cpu.py:338]: On that note, revalidated both XSMM paths. Everything's still good.

**Pull request merged**